model/src/market/proposal.rs

```diff
@@ -86,6 +160,28 @@ impl Proposal {
             .as_ref()
             .ok_or("no previous proposal id".into())
     }
+
+    pub fn chosen_payment_platform(&self) -> Result<String, ErrorMessage> {
```

Comment:
The fact that this method is 20 lines instead of 1 line makes me think there's something seriously wrong with the properties API.